src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobServiceClient;
+import uk.gov.hmcts.reform.blobrouter.data.model.Envelope;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class DuplicateFinder {
+
+    private final BlobServiceClient storageClient;
+    private final EnvelopeService envelopeService;
+
+    public DuplicateFinder(
+        BlobServiceClient storageClient,
+        EnvelopeService envelopeService
+    ) {
+        this.storageClient = storageClient;
+        this.envelopeService = envelopeService;
+    }
+
+    public List<Envelope> findIn(String containerName) {
+        return storageClient
+            .getBlobContainerClient(containerName)
+            .listBlobs()
+            .stream()
+            .map(blob -> envelopeService.findEnvelopes(blob.getName(), containerName))
```

Comment:
may be directly get deleted files?

Reply:
Method name is misleading, this is returning a single envelope.
Fix in https://github.com/hmcts/blob-router-service/pull/264